Source/Core/VideoCommon/VertexManagerBase.cpp

```diff
@@ -595,6 +596,31 @@ void VertexManagerBase::CalculateZSlope(NativeVertexFormat* format)
   m_zslope.dirty = true;
 }
 
+void VertexManagerBase::CalculateBinormals(NativeVertexFormat* format)
+{
+  const PortableVertexDeclaration vert_decl = format->GetVertexDeclaration();
+
+  // Only update the binormal/tangent vertex shader constants if the vertex format lacks binormals
+  // (VertexLoaderManager::binormal_cache gets updated by the vertex loader when binormals are
+  // present, though)
+  if (vert_decl.normals[1].enable)
```

Comment:
Ever a case where `normal[1]` will be set and `normal[2]` will not?

Reply:
No. The 3 options are for the vertex format are:

 * None
 * Normals only
 * Normals + Binormals + Tangents.

Reply:
No.  The existing code doesn't make that super clear though (with separate `VB_HAS_TANGENT`/`VB_HAS_BINORMAL` (`VB_HAS_NRM1` and `VB_HAS_NRM2`) constants, and the array).  There's either 0 normals (`vtx_desc.low.Normal == VertexComponentFormat::NotPresent`), 1 normal (`vtx_desc.low.Normal != VertexComponentFormat::NotPresent` and `vtx_attr.g0.NormalElements == NormalComponentCount::N`), or 3 normals (`vtx_desc.low.Normal != VertexComponentFormat::NotPresent` and `vtx_attr.g0.NormalElements == NormalComponentCount::NBT`).  (There's also the `NormalIndex3` mode, which allows choosing between specifying a single index for the normal, binormal, and tangent, or specifying one index for each of them.)

It also isn't possible for `normal[1]` to be set but `normal[0]` to be unset (which again the logic doesn't make obvious).

Reply:
Thank you both for clarifying!